src/compiler/diagnosticMessages.json

```diff
@@ -2298,6 +2298,10 @@
         "category": "Message",
         "code": 6072
     },
+    "Raise error if two file names in program differ only in case.": {

```

Comment:
Treat file names as case-sensitive, and error if imports and references do not use consistent casing throughout the compilation.